server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/ScheduleService.java

```diff
@@ -789,8 +805,16 @@ public Page<ScheduleTaskListOverview> listScheduleTaskOverviewByScheduleType(@No
                     authenticationFacade.currentOrganizationId(), params.getClusterId()));
         }
         if (StringUtils.isNotEmpty(params.getTenantId())) {
-            params.getDataSourceIds().addAll(connectionService.innerListIdByOrganizationIdAndTenantId(
-                    authenticationFacade.currentOrganizationId(), params.getTenantId()));
+            List<Long> datasourceIds = connectionService.innerListIdByOrganizationIdAndTenantId(
+                    authenticationFacade.currentOrganizationId(), params.getTenantId());
+            if (datasourceIds.isEmpty()) {
+                return Page.empty();
+            }
+            if (params.getDataSourceIds().isEmpty()) {
+                params.getDataSourceIds().addAll(datasourceIds);
+            } else {
+                params.getDataSourceIds().retainAll(datasourceIds);
```

Comment:
why should we retain all?
the familiar logic above does not do this

Reply:
when the filter criteria include both tenantId and clusterId, the union of the two needs to be taken.